### PR TITLE
JAMES-3621 WebAdmin - Clearing content of given mailbox - Task & Route 

### PR DESCRIPTION
**Pull request by vttranlina**
Ref: https://issues.apache.org/jira/browse/JAMES-3621
https://github.com/linagora/james-project/issues/4364

WIP: testcase & document

### CODE REVIEW

File: server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/ClearMailboxContentTaskAdditionalInformationDTO.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    private final String type;
+    private final String username;
+    private final String mailboxName;
+    private final Instant timestamp;
+    private final long messagesSuccessCount;
+    private final long messagesFailCount;
+
+    public ClearMailboxContentTaskAdditionalInformationDTO(String type, String username, String mailboxName, Instant timestamp, long messagesSuccessCount, long messagesFailCount) {
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/ClearMailboxContentTaskAdditionalInformationDTO.java:66]: Can you run this for the distributed server?

I guess we not only need serialization but also deserialisation.

File: server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/UserMailboxesService.java [hunk 2]
```diff
@@ -99,6 +104,33 @@ public boolean testMailboxExists(Username username, MailboxName mailboxName) thr
             .block();
     }
 
+    public Mono<Result> clearMailboxContent(Username username, MailboxName mailboxName, ClearMailboxContentTask.Context context) throws UsersRepositoryException, MailboxException {
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(username);
+        MessageManager messageManager = mailboxManager.getMailbox(MailboxPath.forUser(username, mailboxName.asString()), mailboxSession);
+
+        return Flux.from(messageManager.listMessagesMetadata(MessageRange.all(), mailboxSession))
+            .map(metaData -> metaData.getComposedMessageId().getUid())
+            .map(messageUid -> deleteMessage(messageManager, messageUid, mailboxSession, context))
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/UserMailboxesService.java:113]: deleteMessage is a disguised blocking call, I prefer to wrap it in a Mono.callable scheduled on the Elastic se=cheduler, inside a concatMap

File: server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/service/UserMailboxesServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class UserMailboxesServiceTest {
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/service/UserMailboxesServiceTest.java:25]: Not sure testing the serives is strictly required as you might already have routes test covering this...

**Review by chibenwa — COMMENTED** [review]
**TODO**:

 - Documentation for website and Antora
 - Write a TaskDTO - required by the distributed server to move tasks from one node to the other...
 - Serialisation tests for TaskDTO and AdditionalInformationDTO. See `CreateMissingParentsTaskSerializationTest` + `CreateMissingParentsTaskAdditionalInformationDTOTest`
  - Add something in `RabbitMQWebAdminServerTaskSerializationIntegrationTest` to show that it works ;-)

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Can you squash your fixups please?

**Comment by vttranlina** [comment]
> Can you squash your fixups please?

I think this pr will get some conflict with (https://github.com/apache/james-project/pull/569). So I will fixups & resolve conflict when #569 merged

**Comment by vttranlina** [comment]
Ah no, we can also merge this pr first :D

**Comment by Arsnael** [comment]
> I think this pr will get some conflict with (#569). So I will fixups & resolve conflict when #569 merged

The CI is being capricious these days, I would rather not waste yet time again later.

**Comment by Arsnael** [comment]
Can you rebase and fix conflicts?

**Comment by vttranlina** [comment]
Resolve conflict

**Pull request merged**